src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useContext } from 'react';
+import { Filter } from '../Filter/Filter';
+import { TodosContext } from '../TododsContext/TodosContext';
+import { deleteTodo } from '../../api/todos';
+
+export const Footer: React.FC = () => {
+  const { todos, setTodos } = useContext(TodosContext);
+
+  const countActiveTodos = todos.filter(todo => !todo.completed).length;
+  const countCompletedTodos = todos.some(todo => todo.completed);
+
+  const handleClearCompleted = () => {
+    const newTodos = todos.filter(todo => {
+      if (todo.completed) {
+        deleteTodo(todo.id);
+      }
+
+      return !todo.completed;
+    });
+
+    setTodos(newTodos);
+  };
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {`${countActiveTodos} item${countActiveTodos > 1 ? 's' : ''} left`}
+      </span>
+
+      <Filter />
+      {/* don't show this button if there are no completed todos */}
+      {countCompletedTodos && (
```

Comment:
now when the Clear button doesn't exist filter is not centered - you may use `visibility: hidden` to hide the button but keep the place for it (and disable it if tests fail):

![image](https://github.com/mate-academy/react_todo-app-with-api/assets/86923089/d9a5cc32-271f-4cee-9335-9fd0864190f1)